src/hotspot/share/runtime/thread.cpp

```diff
@@ -1334,6 +1342,7 @@ void NonJavaThread::post_run() {
   unregister_thread_stack_with_NMT();
   // Ensure thread-local-storage is cleared before termination.
   Thread::clear_thread_current();
+  osthread()->set_state(ZOMBIE);
```

Comment:
I'm okay with this change as a debugging marker that the non-JavaThread
has terminated.